umbral/keys.py

```diff
@@ -10,13 +10,78 @@
 from cryptography.hazmat.backends.openssl.ec import (
     _EllipticCurvePublicKey, _EllipticCurvePrivateKey
 )
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
 
 from umbral.config import default_params
-from umbral.point import Point, BigNum
+from umbral.point import Point
+from umbral.bignum import BigNum, hash_to_bn
 from umbral.params import UmbralParameters
 
+from abc import ABC, abstractmethod
+ 
+class AbstractUmbralKeyingMaterial(ABC):
+
+    @abstractmethod
+    def _get_keying_material(self):
+        pass
+
+    def to_bytes(self, password: bytes=None, _scrypt_cost: int=20):
+        """
+        Returns an Umbral private key as a urlsafe base64 encoded string with
+        optional symmetric encryption via nacl's Salsa20-Poly1305 and Scrypt
+        key derivation. If a password is provided, the user must encode it to
+        bytes.
+
+        WARNING: RFC7914 recommends that you use a 2^20 cost value for sensitive
+        files. It is NOT recommended to change the `_scrypt_cost` value unless
+        you know what you are doing.
+        """
+        umbral_get_keying_material = self._get_keying_material()
+
+        if password:
+            salt = os.urandom(16)
+
+            key = Scrypt(
+                salt=salt,
+                length=SecretBox.KEY_SIZE,
+                n=2**_scrypt_cost,
+                r=8,
+                p=1,
+                backend=default_backend()
+            ).derive(password)
+
+            umbral_get_keying_material = SecretBox(key).encrypt(umbral_get_keying_material)
+            umbral_get_keying_material += salt
+
+        encoded_key = base64.urlsafe_b64encode(umbral_get_keying_material)
+        return encoded_key
+
+    @classmethod
+    def _decode_get_keying_material(cls, key_data: bytes, 
```

Comment:
This method name is a little funky. I think I prefer simply `_get_keying_material`.